packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -905,13 +905,11 @@ class _InkResponseState extends State<_InkResponseStateWidget>
     if (widget.radius != oldWidget.radius ||
         widget.highlightShape != oldWidget.highlightShape ||
         widget.borderRadius != oldWidget.borderRadius) {
-      final InkHighlight? hoverHighlight = _highlights[_HighlightType.hover];
-      if (hoverHighlight != null) {
+      if (_highlights case {_HighlightType.hover: final InkHighlight hoverHighlight}) {
```

Comment:
When this proposal lands: https://github.com/dart-lang/language/issues/1201 I feel like this change may make it harder to change those back towards the proposal version, which is less implicit). I'm on the fence with those implicit null checks, it may be a question of getting used to.

Reply:
I should probably revert this, since several other map patterns have been reverted per Andrews suggestion.

Reply:
Agreed. I missed this one in my last review.